src/pybind/mgr/diskprediction/module.py

```diff
@@ -0,0 +1,346 @@
+            return (0,
+                    'success to config device data %s on the device plugin'
+                    % 'not relied' if str_relied.lower() == 'false' else 'relied', 0)
+        except Exception as e:
+            return 0, 'command failed, %s' % str(e), ''
+
+    def _self_test(self, inbuf, cmd):
+        return 0, 'self-test completed', ''
+
+    def _get_predicted_status(self, inbuf, cmd):
+        osd_data = dict()
+        physical_data = dict()
+        try:
+            if not self._prediction_result:
+                for _task in self._tasks:
+                    if isinstance(_task, PredictionTask):
+                        _task.event.set()
+                        break
+            pre_data = self._prediction_result
+            fsid = self.get('mon_map')['fsid']
+            for d_host, d_val in pre_data.get(fsid, {}).iteritems():
+                if "osd.%s" % cmd['osd_id'] in d_val.get("osd", {}).keys():
+                    s_data = d_val['osd']['osd.%s' % cmd['osd_id']]
+                    for dev in s_data.get("physicalDisks", []):
+                        p_data = dev.get('prediction', {})
+                        if not p_data.get('predicted'):
+                            predicted = ''
+                        else:
+                            predicted = datetime.fromtimestamp(int(
+                                p_data.get('predicted')) / (1000 ** 3))
+                        d_data = {
+                            'device': dev.get("diskName"),
+                            'near_failure': p_data.get('near_failure'),
+                            'predicted': str(predicted),
+                            'serial_number': dev.get('serialNumber'),
+                            'disk_wwn': dev.get('diskWWN'),
+                            'attachment': p_data.get('disk_name', '')
+                        }
+                        physical_data[dev.get('diskName')] = d_data
+                    osd_data['osd.%s' % cmd['osd_id']] = dict(
+                        prediction=physical_data)
+                    break
+            if not osd_data:
+                msg = 'not found osd %s predicted data' % cmd['osd_id']
+            else:
+                msg = json.dumps(osd_data, indent=4)
+        except Exception as e:
+            if str(e).find('No such file') >= 0:
+                msg = 'unable to get osd %s predicted data'
+            else:
+                msg = 'unable to get osd %s predicted data, %s' % (cmd['osd_id'], str(e))
+            self.log.error(msg)
+        return 0, msg, ''
+
+    def _config_set(self, inbuf, cmd):
+        self.set_config('diskprediction_server', cmd['server'])
+        self.set_config('diskprediction_user', cmd['user'])
+        self.set_config('diskprediction_password', cmd['password'])
+        if cmd.get('port'):
+            self.set_config('diskprediction_port', cmd['port'])
+        self.show_module_config()
+        return 0, json.dumps(self.config, indent=4), ''
```

Comment:
Setter commands should succeed silently: don't return JSON to the user here.

Reply:
Got it, thanks.